frontend/src/mllike.js

```diff
@@ -43,11 +86,8 @@
             }
         }
 
-        var interfaceSettings = localStorage.getItem('interfaceSettings');
-        var autoIndent = true;
-        if (typeof interfaceSettings === 'string') {
-            autoIndent = !!JSON.parse(interfaceSettings).autoIndent;
-        }
+        /* Use large numbers to see easily possible misuse */
+        const INDENT_OPTIONS = {INDENT: 10, DEDENT: 20, STAY: 30};
 
         function tokenBase(stream, state) {
```

Comment:
The `state.indentHint` should be set in allmost all branches of this function.
Otherwise an out of date value of a previous line will cause wrong indendation.

Reply:
This should be done in line 251 44c15992a414242c9dbbb3853c605632ca0f6b69. This causes that each line starts with the hint INDENT_OPTIONS.STAY.